src/Microsoft.AspNet.OData.Shared/ResourceContext.cs

```diff
@@ -195,6 +196,14 @@ public object GetPropertyValue(string propertyName)
             }
 
             object value;
+            if (SerializerContext.IsDeltaOfT)
+            {
+                if (ResourceInstance is IDelta delta && delta != null && delta.TryGetPropertyValue(propertyName, out value))
```

Comment:
```suggestion
                if (ResourceInstance is IDelta delta && delta.TryGetPropertyValue(propertyName, out value))
```
I believe `ResourceInstance is IDelta delta` will evaluate to false if `ResourceInstance` is not `IDelta`. Therefore there's no need for the `delta != null` conditional check...